Plugins/Mute/Commands/MuteCommand.cs

```diff
@@ -23,20 +26,31 @@ public class MuteCommand : Command
             {
                 Name = translationLookup["COMMANDS_ARGS_PLAYER"],
                 Required = true
+            },
+            new CommandArgument
+            {
+                Name = translationLookup["COMMANDS_ARGS_REASON"],
+                Required = true
             }
         };
     }
-
-    private readonly MuteManager _muteManager = new();
-
+    
     public override async Task ExecuteAsync(GameEvent gameEvent)
     {
-        if (await _muteManager.Mute(gameEvent))
+        var match = Regex.Match(gameEvent.Data, BanRegex);
+        if (match.Success)
         {
-            gameEvent.Origin.Tell($"{_translationLookup["PLUGINS_MUTE_MUTED"]} {gameEvent.Target.Name}");
-            return;
-        }
+            var reason = match.Groups[1].ToString();
 
-        gameEvent.Origin.Tell($"{_translationLookup["PLUGINS_MUTE_UNMUTED"]} {gameEvent.Target.Name}");
+            if (await Plugin.MuteManager.Mute(gameEvent.Owner, gameEvent.Origin, gameEvent.Target, null, reason))
+            {
+                gameEvent.Origin.Tell($"{_translationLookup["PLUGINS_MUTE_MUTED"]} {gameEvent.Target.Name}");
```

Comment:
This is incorrect/outdated usage of the translation. You're not guaranteed every language's word order will be the same.

Translation string (en-US): PLUGINS_MUTE_MUTED => "Muted {{clientName}}"
usage: `gameEvent.Origin.Tell(_translationLookup["PLUGINS_MUTE_MUTED"].FormatExt(gameEvent.Target.Name));`

This decouples the format from code and places it in the translation format. Then the FormatExt is responsible for plugging in the values where needed. It also makes more sense contextually on the translation sheet.